aiida/tools/graph/age_entities.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=unnecessary-pass
+"""Entities for the AiiDA Graph Explorer utility"""
+
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+
+from aiida.orm import Node, Group
+from aiida.orm.querybuilder import QueryBuilder
+from aiida.orm.utils.links import LinkQuadruple
+
+# In order to check valid classes that are implemented:
+VALID_ENTITY_CLASSES = (Node, Group)
+
+# Edges are defined and returned as namedtuples
+GroupNodeEdge = namedtuple('GroupNodeEdge', ['node_id', 'group_id'])
+
+
+class AbstractSetContainer(metaclass=ABCMeta):
+    """This class defines a set that can be speciaized to contain either nodes (aiida nodes )"""
+
+    @abstractmethod
+    def _check_self_and_other(self, other):
+        """
+        Utility function. When called, will check whether self and other instance
+        are compatible. Same aiida classes, same identifiers, length of tuples...
+        """
```

Comment:
How do you handle inheritance with this? For example, this abstract method has to be implemented in each subclass: do I copy the exact same docstring in each of them? Even if the checks are different and I describe them in each docstring, the `:param:`, for example, would still be repeated. Do exclude that part from the "sub-docstrings" or do I repeat it?

Reply:
**Update:** So, apparently the "docbuilder" will identify abstract method implementation in classes and will automatically copy the abstract method's docstring. If I put any docstring in the class implementation, it will overwrite the "inherited" one. So I'll put the docstring in the abstract methods and leave the implemented ones empty. Let me know you thoughts on this.